config/webpack/webpack.desktop.ts

```diff
@@ -1,29 +1,29 @@
-const path = require('path');
-const webpack = require('webpack');
-const _ = require('underscore');
-
-const desktopDependencies = require('../../desktop/package.json').dependencies;
-const getCommonConfig = require('./webpack.common');
+/* eslint-disable @typescript-eslint/naming-convention */
+import path from 'path';
+import type {Configuration} from 'webpack';
+import webpack from 'webpack';
+// eslint-disable-next-line @dword-design/import-alias/prefer-alias, import/no-relative-packages -- alias imports don't work for webpack
+import {dependencies as desktopDependencies} from '../../desktop/package.json';
+import type Env from './types';
+import getCommonConfig from './webpack.common';
 
 /**
  * Desktop creates 2 configurations in parallel
  * 1. electron-main - the core that serves the app content
  * 2. web - the app content that would be rendered in electron
  * Everything is placed in desktop/dist and ready for packaging
- * @param {Object} env
- * @returns {webpack.Configuration[]}
  */
-module.exports = (env) => {
+const getConfig = (env: Env): Configuration[] => {
     const rendererConfig = getCommonConfig({...env, platform: 'desktop'});
     const outputPath = path.resolve(__dirname, '../../desktop/dist');
 
     rendererConfig.name = 'renderer';
-    rendererConfig.output.path = path.join(outputPath, 'www');
+    (rendererConfig.output ??= {}).path = path.join(outputPath, 'www');
 
     // Expose react-native-config to desktop-main
-    const definePlugin = _.find(rendererConfig.plugins, (plugin) => plugin.constructor === webpack.DefinePlugin);
+    const definePlugin = rendererConfig.plugins?.find((plugin) => plugin?.constructor === webpack.DefinePlugin);
```

Comment:
Do we need the `?` in `plugin?.constructor`?

Reply:
Yes, `plugin` might be **null** or **undefined**

Reply:
Interesting. I quickly searched for the type of `rendererConfig.plugins` to try to figure this out and didn't find it, and I was assuming that if we can iterate over `rendererConfig.plugins`, then the items in it wouldn't be null or undefined